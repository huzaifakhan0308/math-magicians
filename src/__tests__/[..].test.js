import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/calculator'
import Home from '../components/home'
import Navbar from '../components/navbar'
import Quote from '../components/quote'
import Calculate from '../logic/calculate'
import { BrowserRouter as Router } from 'react-router-dom';

describe('calculator functions', () => {

  it('addition 5 + 5 to equal 10', () => {
    const object = {
      total: '5',
      next: '5',
      operation: '+',
    }
    const newObj = Calculate(object, '');
    expect(newObj.total).toBe("10")
  });

  it('subtraction 5 - 5 to equal 0', () => {
    const object = {
      total: '5',
      next: '5',
      operation: '-',
    }
    const newObj = Calculate(object, '');
    expect(newObj.total).toBe("0")
  });

  it('multiplication 5 x 5 to equal 25', () => {
    const object = {
      total: '5',
      next: '5',
      operation: 'x',
    }
    const newObj = Calculate(object, '');
    expect(newObj.total).toBe("25")
  });

  it('division 5 รท 5 to equal 1', () => {
    const object = {
      total: '5',
      next: '5',
      operation: 'รท',
    }
    const newObj = Calculate(object, '');
    expect(newObj.total).toBe("1")
  });

  it('percentage 5 % 5 to equal 0', () => {
    const object = {
      total: '5',
      next: '5',
      operation: '%',
    }
    const newObj = Calculate(object, '');
    expect(newObj.total).toBe("0")
  });

  it('equal function', () => {
    const object = {
      total: '5',
      next: '5',
      operation: '+',
    }
    const newObj = Calculate(object, '=');
    expect(newObj.total).toBe('10')
  });

  it('AC function', () => {
    const object = {
      total: '5',
      next: '5',
      operation: '%',
    }
    const newObj = Calculate(object, 'AC');
    expect(newObj.total).toBe(null)
  });

  it('+/- function', () => {
    const object = {
      total: '5',
      next: '5',
      operation: '+',
    }
    const newObj = Calculate(object, '+/-');
    expect(newObj.total).toBe('5')
  });
})

describe('components',()=>{
  it('calculator.jsx should render correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('home.jsx should render correctly', () => {
    const tree = renderer.create(<Home />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('navbar.jsx should render correctly', () => {
    const tree = renderer.create(
      <Router>
        <Navbar />
      </Router>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('navbar.jsx should render correctly', () => {
    const tree = renderer.create(<Quote />).toJSON();

    expect(tree).toMatchSnapshot();
  });
})