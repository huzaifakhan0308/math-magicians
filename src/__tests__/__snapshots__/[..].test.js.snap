// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components calculator.jsx should render correctly 1`] = `
<div
  className="calculator-page-container"
>
  <header
    className="calculator-header"
  >
    Simple Calculator
  </header>
  <main>
    <input
      className="calculator-input"
      onChange={[Function]}
      placeholder="0"
      type="text"
      value=""
    />
    <div
      className="buttons-container"
    >
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="buttons orange-btn"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="buttons orange-btn"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="buttons orange-btn"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="buttons orange-btn"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="button-0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="buttons orange-btn"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </main>
</div>
`;

exports[`components home.jsx should render correctly 1`] = `
<div
  className="home-page-container"
>
  <header
    className="home-header"
  >
    Welcome to our page
  </header>
  <div
    className="home-p"
  >
    <p>
      Lorem ipsum dolor sit, amet consectetur adipisicing elit. Ab libero deleniti neque? Asperiores, molestiae at ducimus quidem dignissimos, possimus distinctio rem nam consectetur dolores corporis voluptates libero ad vero! Enim accusamus in laudantium quod ducimus est porro voluptates sunt provident praesentium asperiores sed soluta cum nesciunt eius, blanditiis quaerat tempora quidem,adipisci fuga laboriosam amet, necessitatibus perferendis. Deleniti, iste quod.
    </p>
    <br />
    <p>
      Lorem ipsum dolor sit, amet consectetur adipisicing elit. Ab libero deleniti neque? Asperiores, molestiae at ducimus quidem dignissimos, possimus distinctio rem nam consectetur dolores corporis voluptates libero ad vero! Enim accusamus in laudantium quod ducimus est porro voluptates sunt provident praesentium asperiores sed soluta cum nesciunt eius.
    </p>
  </div>
</div>
`;

exports[`components navbar.jsx should render correctly 1`] = `
<nav>
  <header>
    Math Magicians
  </header>
  <ul>
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;

exports[`components navbar.jsx should render correctly 2`] = `
<div
  className="qoute-page-container"
>
  Mathematics is not about numbers, equations, computations or algorithms; it is about understanding. -William Paul Thurston
</div>
`;
